### PR TITLE
[MNO-436] notify user of changes

### PR DESCRIPTION
**Pull request by clemthenem**
- Notify users when password is updated.
- Sends an email to the old email when email changes.
- Sends an email to the new email when email changes.

Added specs (omg...)

### CODE REVIEW

File: core/spec/assertions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'active_support/test_case'
```
Comment (ouranos) [inline core/spec/assertions_helper.rb]: No need for ActiveSupport::TestCase helper

File: api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb [hunk 2]
```diff
@@ -32,12 +32,21 @@ def default_sender
   #   :confirmation_link
   #
   def confirmation_instructions(record, token, opts={})
-    template = record.confirmed? && record.unconfirmed_email? ? 'reconfirmation-instructions' : 'confirmation-instructions'
+    update_email = record.confirmed? && record.unconfirmed_email?
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb]: Could you add more comments here to explain what's going on?
Either on the method doc or inline

**Comment by clemthenem** [comment]
Hey @ouranos! 
I did the changed requested.
Let me know if you need anything else from me on this one.
Cheers

File: api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb (hunk 2 above)
Comment (clemthenem) [inline api/lib/mno_enterprise/concerns/mailers/system_notification_mailer.rb]: Just finished to do that on the method doc

**Pull request merged**